internal/core/src/index/IndexFactory.cpp

```diff
@@ -95,8 +95,26 @@ IndexFactory::CreateVectorIndex(
     }
 
     // create mem index
-    return std::make_unique<VectorMemIndex>(
-        index_type, metric_type, version, file_manager_context);
+    else {
```

Comment:
the `else` is away from the `if` block too far, like to move it up or remove this `else` block

Reply:
Ok, I'll fix it.